src/monster.cpp

```diff
@@ -1384,55 +1393,51 @@ int monster::stability_roll() const
     return stability;
 }
 
-int monster::get_dodge() const
-{
-    if (has_effect( effect_downed)) {
-        return 0;
-    }
-    int ret = type->sk_dodge;
-    if( has_effect( effect_lightsnare ) || has_effect( effect_heavysnare ) || has_effect( effect_beartrap ) || has_effect( effect_tied ) ) {
-        ret /= 2;
-    }
-    if (moves <= 0 - 100 - get_speed()) {
-        ret = rng(0, ret);
-    }
-    return ret + get_dodge_bonus();
-}
-
-int monster::get_melee() const
-{
-    return type->melee_skill;
-}
-
-int monster::dodge_roll()
+float monster::get_dodge() const
 {
-    if (has_effect( effect_bouldering)) {
-        if(one_in(type->sk_dodge)) {
-            return 0;
-        }
-    }
-
-    int numdice = get_dodge();
-
-    switch (type->size) {
+    float ret = 0.0f;
+    switch( type->size ) {
```

Comment:
Definition should probably be moved to `JSON` with default value set on load dependent upon size. Doesn't need to be this PR